gufe/settings/models.py

```diff
@@ -0,0 +1,174 @@
+"""
+Pydantic models used for storing settings.
+"""
+
+from datetime import date
+from typing import Literal, Union
+
+from pydantic import BaseModel, Extra, FilePath, PositiveFloat
+from typing_extensions import TypedDict
+
+
+class SettingsBaseModel(BaseModel):
+    """
+    Settings and modifications we want for all settings classes.
+    """
+
+    class Config:
+        extra = Extra.forbid
+        # Immutability in python is never strict.
+        # If developers are determined/stupid
+        # they can always modify a so-called "immutable" object.
+        # https://pydantic-docs.helpmanual.io/usage/models/#faux-immutability
+        allow_mutation = False
+        arbitrary_types_allowed = True
+        smart_union = True
+
+
+class vdWScale(TypedDict):
+    scale12: float
+    scale13: float
+    scale14: float
+    scale15: float
+
+
+class ElectrostaticScale(TypedDict):
+    scale12: float
+    scale13: float
+    scale14: float
+    scale15: float
+
+
+class VdWSettings(SettingsBaseModel):
+    """
+    Settings for van der Waals force
+    """
+
+    combining_rules: Literal["Lorentz-Berthelot"]
+    potential: Literal["Lennard-Jones-12-6"]
+    scale: vdWScale
+    long_range_dispersion: Literal["isotropic"]
+    cutoff: float  # angstrom
+    switch_width: float  # angstrom
+    method: Literal["cutoff"]
+
+
+class ElectrostaticSettings(SettingsBaseModel):
+    """
+    Settings for electrostatics
+    """
+
+    # Tricky since this allows functions https://openforcefield.github.io/standards/standards/smirnoff/#electrostatics
+    periodic_potential: Union[Literal["Ewald3D-ConductingBoundary"], str]
+    nonperiodic_potential: Union[Literal["Coulomb"], str]
+    exception_potential: Union[Literal["Coulomb"], str]
+    scale: ElectrostaticScale
+    cutoff: Union[float, None]
+    switch_width: Union[float, None]
+    solvent_dielectric: Union[float, None]
```

Comment:
`Union[float, None]` is better stated as `Optional[float]`

Reply:
These will all change once I add in units, but that is good to know!